src/main/scala/org/broadinstitute/dsde/agora/server/business/PermissionBusiness.scala

```diff
@@ -1,72 +1,114 @@
+    checkSameRequester(requester, userToRemove)
+    authorizeNamespaceRequester(getNamespaceACLs(entity, requester), entity, requester)
     NamespacePermissionsClient.deleteNamespacePermission(entity, userToRemove)
   }
 
-  def listEntityPermissions(entity: AgoraEntity, requester: String) = {
-    authorizeEntityRequester(entity, requester)
+  def listEntityPermissions(entity: AgoraEntity, requester: String): Seq[AccessControl] = {
+    authorizeEntityRequester(getEntityACLs(entity, requester), entity, requester)
     AgoraEntityPermissionsClient.listEntityPermissions(entity)
   }
 
-  def insertEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl) = {
-    authorizeEntityRequester(entity, requester)
+  def insertEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl): Int = {
+    // Inserts can result in an edit in the DAO layer
+    val entityACLs = getEntityACLs(entity, requester)
+    checkSameRequesterAndPermissions(entityACLs.find(acl => acl.user.equals(requester)), accessObject)
+    authorizeEntityRequester(entityACLs, entity, requester)
     AgoraEntityPermissionsClient.insertEntityPermission(entity, accessObject)
   }
 
-  def batchEntityPermission(entity: AgoraEntity, requester: String, accessObjectList: List[AccessControl]) = {
+  def batchEntityPermission(entity: AgoraEntity, requester: String, accessObjectList: List[AccessControl]): Int = {
     accessObjectList.map(accessObject => insertEntityPermission(entity, requester, accessObject)).sum
   }
 
-  def editEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl) = {
-    authorizeEntityRequester(entity, requester)
+  def editEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl): Int = {
+    val entityACLs = getEntityACLs(entity, requester)
+    checkSameRequesterAndPermissions(entityACLs.find(acl => acl.user.equals(requester)), accessObject)
+    authorizeEntityRequester(entityACLs, entity, requester)
     AgoraEntityPermissionsClient.editEntityPermission(entity, accessObject)
 
   }
 
-  def deleteEntityPermission(entity: AgoraEntity, requester: String, userToRemove: String) = {
-    authorizeEntityRequester(entity, requester)
+  def deleteEntityPermission(entity: AgoraEntity, requester: String, userToRemove: String): Int = {
+    checkSameRequester(requester, userToRemove)
+    authorizeEntityRequester(getEntityACLs(entity, requester), entity, requester)
     AgoraEntityPermissionsClient.deleteEntityPermission(entity, userToRemove)
   }
 
-  def authorizeNamespaceRequester(entity: AgoraEntity, requester: String) = {
-    if (!NamespacePermissionsClient.getNamespacePermission(entity, requester).canManage &&
-        !NamespacePermissionsClient.getNamespacePermission(entity, "public").canManage)
-      throw new NamespaceAuthorizationException(AgoraPermissions(Manage), entity, requester)
+
```

Comment:
The `authorizeNamespaceRequester` and `authorizeEntityRequester` methods previously made two db queries. Each is refactored so that we can make a single db call for the relevant ACLs and pass those in.